src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/MemoryHandles.java

```diff
@@ -265,4 +333,20 @@ private static long carrierSize(Class<?> carrier) {
         long bitsAlignment = Math.max(8, Wrapper.forPrimitiveType(carrier).bitWidth());
         return Utils.bitsToBytesOrThrow(bitsAlignment, IllegalStateException::new);
     }
+
+    private static MemoryAddress longToAddress(long value) {
+        return MemoryAddress.ofLong(value);
+    }
+
+    private static long addressToLong(MemoryAddress value) {
+        return ((MemoryAddressImpl)value).unsafeGetOffset();
```

Comment:
Maybe this should use MemoryAddressImpl::addressof, since that also checks for heap addresses. I don't think we want to allow serializing heap addresses?

Reply:
yeah - that's what it should do